### PR TITLE
Improve timeout behaviour in tests

### PR DESCRIPTION
**Pull request by Porges**
At the moment we allow the tests to run as long as possible governed by the overall test timeout (deadline).

Instead, we should have a lower per-operation timeout that can be overridden by the deadline if we are close to hitting it.

### CODE REVIEW

File: v2/internal/testcommon/kube_per_test_context.go [hunk 1]
```diff
@@ -246,19 +247,19 @@ func (tc *KubePerTestContext) Subtest(t *testing.T) *KubePerTestContext {
 	return result
 }
 
-// DefaultTimeoutReplaying is the default timeout for a single operation when replaying.
-var DefaultTimeoutReplaying = 2 * time.Minute
+// OperationTimeoutReplaying is the default timeout for a single operation when replaying.
+var OperationTimeoutReplaying = 30 * time.Second
```
Comment (matthchr) [inline v2/internal/testcommon/kube_per_test_context.go:251]: We might need to leave this at like 1m (I'm not sure how I picked 2m). If it passes at 30s though then awesome let's use that.

**Review by matthchr — APPROVED** [review]

**Review by babbageclunk — APPROVED** [review]

File: v2/internal/testcommon/kube_per_test_context.go (hunk 1 above)
Comment (Porges) [reply v2/internal/testcommon/kube_per_test_context.go:251]: If it takes 30s to achieve the desired state I think something has definitely gone wrong?

**Review by theunrepentantgeek — APPROVED** [review]

File: v2/internal/testcommon/kube_per_test_context.go (hunk 1 above)
Comment (Porges) [reply v2/internal/testcommon/kube_per_test_context.go:251]:
```suggestion
var OperationTimeoutReplaying = 2 * time.Minute
```

Comment (Porges) [reply v2/internal/testcommon/kube_per_test_context.go]: Restored to 2 minutes because apparently the Mongo/SQL ones do take a while.

**Pull request merged**